drivers/SmartThings/zigbee-switch/src/init.lua

```diff
@@ -85,6 +90,55 @@ local device_init = function(self, device)
   end
 end
 
+local function is_mcd_device(device)
+  local components = device.profile.components
+  if type(components) == "table" then
+    local component_count = 0
+    for _, component in pairs(components) do
+        component_count = component_count + 1
+    end
+    if component_count >= 2 then
+      return true
+    else
+      return false
+    end
+  end
+end
+
+local function device_added(driver, device, event)
+  local num_switch_server_eps = 0
+  local main_endpoint = device:get_endpoint(clusters.OnOff.ID)
+  local updated_flag = false
+  if is_mcd_device(device) == false and device.network_type == device_lib.NETWORK_TYPE_ZIGBEE then
+    for _, ep in ipairs(device.zigbee_endpoints) do
```

Comment:
you could probably use this index in the place of `num_switch_server_eps` rather than discard it